server/src/main/java/com/flipkart/varadhi/utils/MessageRequestValidator.java

```diff
@@ -0,0 +1,53 @@
+package com.flipkart.varadhi.utils;
+
+import java.util.Map;
+
+import com.flipkart.varadhi.config.MessageConfiguration;
+import com.flipkart.varadhi.entities.StdHeaders;
+import com.google.common.base.Utf8;
+import com.google.common.collect.Multimap;
+
+public class MessageRequestValidator {
+
+    public static void ensureHeaderSemanticsAndSize(
+        MessageConfiguration msgConfig,
+        Multimap<String, String> requestHeaders,
+        long bodyLength
+    ) {
+        msgConfig.ensureRequiredHeaders(requestHeaders);
+        long headersAndBodySize = 0;
+
+        for (Map.Entry<String, String> entry : requestHeaders.entries()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+
+            if (isIdHeader(msgConfig.stdHeaders(), key)) {
```

Comment:
why only validate header length for msgId and GroupId?

what about other headers?